src/range.rs

```diff
@@ -125,8 +122,8 @@ impl<V: Version> Range<V> {
     /// For example:
     ///    [ (v1, None) ] => [ (start, Some(v1)) ]
     ///    [ (v1, Some(v2)) ] => [ (start, Some(v1)), (v2, None) ]
-    fn negate_segments(start: V, segments: &[Interval<V>]) -> Vec<Interval<V>> {
-        let mut complement_segments = Vec::with_capacity(1 + segments.len());
+    fn negate_segments(start: V, segments: &[Interval<V>]) -> Range<V> {
+        let mut complement_segments = SmallVec::with_capacity(1 + segments.len());
```

Comment:
Negation (complement) of a range result at most into one additional segment. Since this is used a lot, it would be worth trying to extend the smallvec enum to 3 or 4 elements, and initialize this with `SmallVec::empty` instead I think.